### ISSUE TITLE
A problem when monitoring a Switch via SNMP.

### ISSUE THREAD

**Issue by talkyy** [firehol/netdata#2955]
Hi,
I want to use netdata for monitoring some switch in my network via SNMP. I am trying to configure to monitor a switch first but I got a problem. I can't see chart of this switch on dashboard. I also try to push the metrics to Influxdb and use Grafana but I don't see chart, too.

<details>
<summary>Click for my configuration</summary>

1. File `snmp.conf`:

```
{
    "enable_autodetect": false,
    "update_every": 10,
    "servers": [
        {
            "hostname": "172.28.254.2",
            "community": "hcmut",
            "update_every": 10,
            "options": { "timeout": 20000 },
            "charts": {
                "snmp_switch.bandwidth_port": {
                    "title": "Switch Bandwidth for port ",
                    "units": "kilobits/s",
                    "type": "area",
                    "priority": 1,
                    "family": "ports",
                    "multiply_range": [ 1, 24 ],
                    "dimensions": {
                        "in": {
                            "oid": "1.3.6.1.2.1.2.2.1.10.",
                            "algorithm": "incremental",
                            "multiplier": 8,
                            "divisor": 1024,
                            "offset": 0
                        },
                        "out": {
                            "oid": "1.3.6.1.2.1.2.2.1.16.",
                            "algorithm": "incremental",
                            "multiplier": -8,
                            "divisor": 1024,
                            "offset": 0
                        }
                    }
                }
            }
        }
    ]
}
```
2. File `netdata.conf`:

```
[backend]
	# host tags = 
	 enabled = yes
	 data source = as collected
	 type = opentsdb
	 destination = localhost:4242
	 prefix = netdata
	 hostname = netdata-server
	 update every = 10
	 buffer on failures = 10
	 timeout ms = 20000
	 send names instead of ids = yes
	 send charts matching = *
```
3. When I run `/usr/libexec/netdata/plugins.d/node.d.plugin 1 snmp`

```
CHART "snmp_switch.bandwidth_port1" "snmp_switch.bandwidth_port1" "Switch Bandwidth for port 1" "kilobits/s" "ports" "" "area" 50001 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port1
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port2" "snmp_switch.bandwidth_port2" "Switch Bandwidth for port 2" "kilobits/s" "ports" "" "area" 50002 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port2
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port3" "snmp_switch.bandwidth_port3" "Switch Bandwidth for port 3" "kilobits/s" "ports" "" "area" 50003 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port3
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port4" "snmp_switch.bandwidth_port4" "Switch Bandwidth for port 4" "kilobits/s" "ports" "" "area" 50004 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port4
SET in = 1319784277
SET out = 3253352981
END

CHART "snmp_switch.bandwidth_port5" "snmp_switch.bandwidth_port5" "Switch Bandwidth for port 5" "kilobits/s" "ports" "" "area" 50005 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port5
SET in = 593765481
SET out = 4087631153
END

CHART "snmp_switch.bandwidth_port6" "snmp_switch.bandwidth_port6" "Switch Bandwidth for port 6" "kilobits/s" "ports" "" "area" 50006 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port6
SET in = 72168946
SET out = 2679416486
END

CHART "snmp_switch.bandwidth_port7" "snmp_switch.bandwidth_port7" "Switch Bandwidth for port 7" "kilobits/s" "ports" "" "area" 50007 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port7
SET in = 2459847384
SET out = 3883955581
END

CHART "snmp_switch.bandwidth_port8" "snmp_switch.bandwidth_port8" "Switch Bandwidth for port 8" "kilobits/s" "ports" "" "area" 50008 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port8
SET in = 436818597
SET out = 3329605045
END

CHART "snmp_switch.bandwidth_port9" "snmp_switch.bandwidth_port9" "Switch Bandwidth for port 9" "kilobits/s" "ports" "" "area" 50009 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port9
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port10" "snmp_switch.bandwidth_port10" "Switch Bandwidth for port 10" "kilobits/s" "ports" "" "area" 50010 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port10
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port11" "snmp_switch.bandwidth_port11" "Switch Bandwidth for port 11" "kilobits/s" "ports" "" "area" 50011 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port11
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port12" "snmp_switch.bandwidth_port12" "Switch Bandwidth for port 12" "kilobits/s" "ports" "" "area" 50012 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port12
SET in = 116876798
SET out = 896732214
END

CHART "snmp_switch.bandwidth_port13" "snmp_switch.bandwidth_port13" "Switch Bandwidth for port 13" "kilobits/s" "ports" "" "area" 50013 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port13
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port14" "snmp_switch.bandwidth_port14" "Switch Bandwidth for port 14" "kilobits/s" "ports" "" "area" 50014 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port14
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port15" "snmp_switch.bandwidth_port15" "Switch Bandwidth for port 15" "kilobits/s" "ports" "" "area" 50015 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port15
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port16" "snmp_switch.bandwidth_port16" "Switch Bandwidth for port 16" "kilobits/s" "ports" "" "area" 50016 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port16
SET in = 354443112
SET out = 1469483087
END

CHART "snmp_switch.bandwidth_port17" "snmp_switch.bandwidth_port17" "Switch Bandwidth for port 17" "kilobits/s" "ports" "" "area" 50017 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port17
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port18" "snmp_switch.bandwidth_port18" "Switch Bandwidth for port 18" "kilobits/s" "ports" "" "area" 50018 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port18
SET in = 704418420
SET out = 3306302808
END

CHART "snmp_switch.bandwidth_port19" "snmp_switch.bandwidth_port19" "Switch Bandwidth for port 19" "kilobits/s" "ports" "" "area" 50019 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port19
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port20" "snmp_switch.bandwidth_port20" "Switch Bandwidth for port 20" "kilobits/s" "ports" "" "area" 50020 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port20
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port21" "snmp_switch.bandwidth_port21" "Switch Bandwidth for port 21" "kilobits/s" "ports" "" "area" 50021 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port21
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port22" "snmp_switch.bandwidth_port22" "Switch Bandwidth for port 22" "kilobits/s" "ports" "" "area" 50022 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port22
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port23" "snmp_switch.bandwidth_port23" "Switch Bandwidth for port 23" "kilobits/s" "ports" "" "area" 50023 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port23
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port24" "snmp_switch.bandwidth_port24" "Switch Bandwidth for port 24" "kilobits/s" "ports" "" "area" 50024 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port24
SET in = 460543611
SET out = 1378853007
END

```
</details>

---

How can I do to show the charts that are collected via SNMP? #

**Comment by ktsaou** [comment]
Hi,

could you please post the output of this:

```
grep -Ei "(snmp|node\.d\.plugin)" /var/log/netdata/error.log
```

**Comment by talkyy** [comment]
Hi,
There is no output when I run that command. My final purpose is by using netdata to collect metrics of switch via SNMP, push all metrics to Influxdb and use Grafana to display. I think maybe I forgot something when I configured these files. According document in wiki, I configure file `snmp.conf` and restart netdata, I can see the charts on dashboard (`http://localhost:19999`). Is it right?
Thank you.

**Comment by ktsaou** [comment]
Yes, this is how it works.

But something is missing on your setup.

Please run this:

```sh
# become user netdata
sudo su -s /bin/bash netdata

# run the snmp plugin in debug mode, by hand
/usr/libexec/netdata/plugins.d/node.d.plugin debug 1 snmp
```

Could you please post the output?

**Comment by talkyy** [comment]
Yes. I ran that command, it takes over 800 lines (I edited `"multiply_range": [ 1, 24 ]` to `"multiply_range": [ 1, 2 ]` in the file `snmp.conf` so that netdata only collects port1 and port2 instead of 24 ports). I think I shouldn't post all output in this comment so I copy all them into a file. Could you please view the output in the file that I attached?
Thanks a lot.
[Output_SNMP_Debug.txt](https://github.com/firehol/netdata/files/1448591/Output_SNMP_Debug.txt)

**Comment by talkyy** [comment]
Hi, 
I run again the command `grep -Ei "(snmp|node\.d\.plugin)" /var/log/netdata/error.log` and here is output:
```
2017-11-07 14:49:09: netdata INFO : PLUGINSD: '/usr/libexec/netdata/plugins.d/node.d.plugin' running on pid 15341
2017-11-07 14:49:10: node.d.plugin: ERROR: Cannot load module's stiebeleltron config from /etc/netdata/node.d/stiebeleltron.conf exception: Error: ENOENT: no such file or directory, open '/etc/netdata/node.d/stiebeleltron.conf', using internal defaults.
2017-11-07 14:49:10: node.d.plugin: ERROR: Cannot load module's sma_webbox config from /etc/netdata/node.d/sma_webbox.conf exception: Error: ENOENT: no such file or directory, open '/etc/netdata/node.d/sma_webbox.conf', using internal defaults.
2017-11-07 14:49:10: node.d.plugin: ERROR: Cannot load module's named config from /etc/netdata/node.d/named.conf exception: Error: ENOENT: no such file or directory, open '/etc/netdata/node.d/named.conf', using internal defaults.
2017-11-07 14:49:10: node.d.plugin: ERROR: Cannot load module's fronius config from /etc/netdata/node.d/fronius.conf exception: Error: ENOENT: no such file or directory, open '/etc/netdata/node.d/fronius.conf', using internal defaults.
2017-11-07 14:49:10: netdata ERROR: PLUGINSD: /usr/libexec/netdata/plugins.d/node.d.plugin : read failed.
2017-11-07 14:49:10: netdata ERROR: PLUGINSD: plugin '/usr/libexec/netdata/plugins.d/node.d.plugin' disconnected.
2017-11-07 14:49:10: netdata INFO : PLUGINSD: '/usr/libexec/netdata/plugins.d/node.d.plugin' on pid 15341 stopped after 0 successful data collections (ENDs).
2017-11-07 14:49:10: netdata ERROR: PLUGINSD: '/usr/libexec/netdata/plugins.d/node.d.plugin' exited with error code 1. Disabling it.
2017-11-07 14:49:10: netdata INFO : PLUGINSD: '/usr/libexec/netdata/plugins.d/node.d.plugin' thread exiting
```

**Comment by ktsaou** [comment]
Well, it seems that it runs when you run it by hand, but it fails when netdata runs it.

Could you please check if you have any logs between these 2 lines:

```
2017-11-07 14:49:10: node.d.plugin: ERROR: Cannot load module's fronius config from /etc/netdata/node.d/fronius.conf exception: Error: ENOENT: no such file or directory, open '/etc/netdata/node.d/fronius.conf', using internal defaults.
2017-11-07 14:49:10: netdata ERROR: PLUGINSD: /usr/libexec/netdata/plugins.d/node.d.plugin : read failed.
```

For some reason netdata cannot read the plugin output, so I think there may be an exception right between these 2 lines.

**Comment by talkyy** [comment]
It seems that it needs 4 files in directory `/etc/netdata/node.d/` such as `stiebeleltron.conf`, `sma_webbox.conf`, `named.conf`,` fronius.conf`. I checked and these files are missing. Is it the problem?

**Comment by ktsaou** [comment]
> It seems that it needs 4 files in directory /etc/netdata/node.d/ such as stiebeleltron.conf, sma_webbox.conf, named.conf,fronius.conf. I checked and these files are missing. Is it the problem?

No, it is ok if these are missing. When you run it by hand it complaints about these files too and it worked for SNMP.

**Comment by talkyy** [comment]
I have just checked and there is nothing between these 2 lines:
```
2017-11-07 20:43:21: netdata INFO : PLUGINSD: '/usr/libexec/netdata/plugins.d/node.d.plugin' running on pid 8377
2017-11-07 20:43:21: node.d.plugin: ERROR: Cannot load module's stiebeleltron config from /etc/netdata/node.d/stiebeleltron.conf exception: Error: ENOENT: no such file or directory, open '/etc/netdata/node.d/stiebeleltron.conf', using internal defaults.
2017-11-07 20:43:21: node.d.plugin: ERROR: Cannot load module's sma_webbox config from /etc/netdata/node.d/sma_webbox.conf exception: Error: ENOENT: no such file or directory, open '/etc/netdata/node.d/sma_webbox.conf', using internal defaults.
2017-11-07 20:43:21: node.d.plugin: ERROR: Cannot load module's named config from /etc/netdata/node.d/named.conf exception: Error: ENOENT: no such file or directory, open '/etc/netdata/node.d/named.conf', using internal defaults.
2017-11-07 20:43:21: node.d.plugin: ERROR: Cannot load module's fronius config from /etc/netdata/node.d/fronius.conf exception: Error: ENOENT: no such file or directory, open '/etc/netdata/node.d/fronius.conf', using internal defaults.
2017-11-07 20:43:21: netdata ERROR: PLUGINSD: /usr/libexec/netdata/plugins.d/node.d.plugin : read failed.
2017-11-07 20:43:21: netdata ERROR: PLUGINSD: plugin '/usr/libexec/netdata/plugins.d/node.d.plugin' disconnected.
2017-11-07 20:43:21: netdata INFO : PLUGINSD: '/usr/libexec/netdata/plugins.d/node.d.plugin' on pid 8377 stopped after 0 successful data collections (ENDs).
2017-11-07 20:43:21: netdata ERROR: PLUGINSD: '/usr/libexec/netdata/plugins.d/node.d.plugin' exited with error code 1. Disabling it.
2017-11-07 20:43:21: netdata INFO : PLUGINSD: '/usr/libexec/netdata/plugins.d/node.d.plugin' thread exiting
```
The output also has many lines of this:
```
2017-11-07 19:17:39: netdata ERROR: File /var/cache/netdata/apps.cpu/node.d.plugin.db is too old (last collected 17482 seconds ago, but the database is 3996 seconds). Clearing it.
2017-11-07 19:17:39: netdata ERROR: File /var/cache/netdata/apps.mem/node.d.plugin.db is too old (last collected 17482 seconds ago, but the database is 3996 seconds). Clearing it.
```
Does netdata needs any packets to use `node.d.plugin`? I installed `nodejs` by this command:
`sudo apt install nodejs`
Is it enough?

Thanks a lot.

**Comment by ktsaou** [comment]
> Does netdata needs any packets to use node.d.plugin? I installed nodejs by this command:
> sudo apt install nodejs
> Is it enough?

yes, it should.

Could you please try to run the plugin as user `netdata`:

```sh
# become user netdata
sudo su -s /bin/bash netdata

# run the plugin in debug mode
/usr/libexec/netdata/plugins.d/node.d.plugin debug 1 snmp
```

Does it run now?

**Comment by talkyy** [comment]
Yes, I try to run and it run.
Here is the output:
[Output_.txt](https://github.com/firehol/netdata/files/1452649/Output_.txt)

**Comment by ktsaou** [comment]
ok, this is spooky... :)

It runs perfectly by hand, but when netdata runs it, it exits with code 1 (fail).

```
2017-11-07 20:43:21: netdata ERROR: PLUGINSD: /usr/libexec/netdata/plugins.d/node.d.plugin : read failed.
2017-11-07 20:43:21: netdata ERROR: PLUGINSD: plugin '/usr/libexec/netdata/plugins.d/node.d.plugin' disconnected.
2017-11-07 20:43:21: netdata INFO : PLUGINSD: '/usr/libexec/netdata/plugins.d/node.d.plugin' on pid 8377 stopped after 0 successful data collections (ENDs).
2017-11-07 20:43:21: netdata ERROR: PLUGINSD: '/usr/libexec/netdata/plugins.d/node.d.plugin' exited with error code 1. Disabling it.
2017-11-07 20:43:21: netdata INFO : PLUGINSD: '/usr/libexec/netdata/plugins.d/node.d.plugin' thread exiting
```

Could you please post this:

```
ls -ld /etc/netdata/node.d
ls -l /etc/netdata/node.d/snmp.conf
```

Which distro is this? I'll try to setup a VM with it to test it.

**Comment by talkyy** [comment]
Here is the output of the first command:
`drwxrwxr-x 2 netdata netdata 4096 Th11  7 18:58 /etc/netdata/node.d
`
And the other one:
`-rw-rw---- 1 netdata netdata 1293 Th11  7 14:10 /etc/netdata/node.d/snmp.conf`

**Comment by talkyy** [comment]
Well, I have rebooted machine and the list of snmp device appear in the sub-menu but the field of charts show empty without error code 1. Then I restarted netdata.service and refreshed dashboard, the list disappear and the error code 1 appear again.

**Comment by ktsaou** [comment]
ok, let's give it another try:

1. delete `/var/log/netdata/error.log` by running `sudo rm /var/log/netdata/error.log`
2. restart netdata with `systemctl restart netdata`
3. wait 1 minute
4. post `/var/log/netdata/error.log` here

**Comment by talkyy** [comment]
Yes, this is `error.log`:
[error.log](https://github.com/firehol/netdata/files/1460486/error.log)
If it isn't helpful, maybe I should re-install netdata. :)

**Comment by ktsaou** [comment]
It is a bug:

```
2017-11-10 14:38:12: node.d.plugin: ERROR: Cannot load module's fronius config from /etc/netdata/node.d/fronius.conf exception: Error: ENOENT: no such file or directory, open '/etc/netdata/node.d/fronius.conf', using internal defaults.
undefined:1
<!DOCTYPE html>
^

SyntaxError: Unexpected token <
    at Object.parse (native)
    at named.processResponse (/usr/libexec/netdata/node.d/named.node.js:201:26)
    at /usr/libexec/netdata/node.d/node_modules/netdata.js:248:17
    at IncomingMessage.<anonymous> (/usr/libexec/netdata/node.d/node_modules/netdata.js:113:36)
    at emitNone (events.js:72:20)
    at IncomingMessage.emit (events.js:166:7)
    at endReadableNT (_stream_readable.js:905:12)
    at nextTickCallbackWith2Args (node.js:441:9)
    at process._tickCallback (node.js:355:17)
2017-11-10 14:38:12: netdata ERROR: PLUGINSD: /usr/libexec/netdata/plugins.d/node.d.plugin : read failed.
```

The `named` module crashes the whole plugin.
I'll fix it later today.

**Comment by talkyy** [comment]
Yes, I'm looking forward to your fixing soon.

**Comment by ktsaou** [comment]
Fixed in PR #2998 
Can you test it?

**Comment by talkyy** [comment]
Yes, it works! :)
But there is a thing. The charts have no metrics. The others charts still work but the charts of SNMP don's work. They stop at the time I start or restart netdata. Here is the image of SNMP on dashboard:
![image](https://user-images.githubusercontent.com/26269505/32685734-b3326c2a-c6c9-11e7-9909-a00a9506e762.png)
Thanks for your help.

**Comment by ktsaou** [comment]
hm... so it generated the charts but it does not push values.
Try this by hand:

```sh
# become user netdata
sudo su -s /bin/bash netdata

# run the plugin in debug mode
/usr/libexec/netdata/plugins.d/node.d.plugin 1
```

When you run it just for `snmp` it worked. The above commands runs the plugin the same way netdata runs it. Does it roll (generating data all the time)?

**Comment by talkyy** [comment]
Oh, it only gets metrics one time (I mean it does not roll). Here is `error.log`, maybe it will help:
[error.log](https://github.com/firehol/netdata/files/1463786/error.log)
I will check again and tell you if there are something new.

**Comment by ktsaou** [comment]
Unfortunately this error.log does not provide any useful info.
Could you please the command given above?

**Comment by talkyy** [comment]
Yes. I configured that netdata monitors port1 and port2 of switch. Here is output when I run the command `/usr/libexec/netdata/plugins.d/node.d.plugin 1` with netdata user:
```
2017-11-12 10:30:30: node.d.plugin: ERROR: Cannot load module's stiebeleltron config from /usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/stiebeleltron.conf exception: Error: ENOENT: no such file or directory, open '/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/stiebeleltron.conf', using internal defaults.
2017-11-12 10:30:30: node.d.plugin: ERROR: Cannot load module's sma_webbox config from /usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/sma_webbox.conf exception: Error: ENOENT: no such file or directory, open '/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/sma_webbox.conf', using internal defaults.
2017-11-12 10:30:30: node.d.plugin: ERROR: Cannot load module's named config from /usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/named.conf exception: Error: ENOENT: no such file or directory, open '/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/named.conf', using internal defaults.
2017-11-12 10:30:30: node.d.plugin: ERROR: Cannot load module's fronius config from /usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/fronius.conf exception: Error: ENOENT: no such file or directory, open '/usr/libexec/netdata/plugins.d/../../../../etc/netdata/node.d/fronius.conf', using internal defaults.
CHART "snmp_switch.bandwidth_port1" "snmp_switch.bandwidth_port1" "Switch Bandwidth for port 1" "kilobits/s" "snmp_device" "" "area" 50001 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port1
SET in = 64
SET out = 64
END

CHART "snmp_switch.bandwidth_port2" "snmp_switch.bandwidth_port2" "Switch Bandwidth for port 2" "kilobits/s" "snmp_device" "" "area" 50002 10
DIMENSION "in" "in" "incremental" 8 1024
DIMENSION "out" "out" "incremental" -8 1024
BEGIN snmp_switch.bandwidth_port2
SET in = 64
SET out = 64
END
```
It does not generate new values. 

And this is output when I run it in debug mode:

[Debug_mode.txt](https://github.com/firehol/netdata/files/1464350/Debug_mode.txt)

**Comment by ktsaou** [comment]
ok, please run it in debug mode like this and leave it running at least 20 seconds after the output you gave above. You have configured it to collect metrics every 10 seconds, but the debug output has only 2 seconds after the first iteration, so I can't see what is happening at the next one:

```
# become user netdata
sudo su -s /bin/bash netdata

# run the plugin in debug mode
/usr/libexec/netdata/plugins.d/node.d.plugin 1 debug
```

**Comment by talkyy** [comment]
Oh, my mistake. Here is the output:
[ouptut.txt](https://github.com/firehol/netdata/files/1464690/ouptut.txt)
I think it can not load modules so the process has to wait.

**Comment by talkyy** [comment]
Oh, my mistake. Here is output:
[ouptut.txt](https://github.com/firehol/netdata/files/1464702/ouptut.txt)

**Comment by talkyy** [comment]
Hi, here is output of the command `/usr/libexec/netdata/plugins.d/node.d.plugin debug 1`:
[output_noded_13_11.txt](https://github.com/firehol/netdata/files/1466661/output_noded_13_11.txt)
I also run the command `grep -Ei "(snmp|node\.d\.plugin)" /var/log/netdata/error.log` after I removed `error.log` and restarted netdata, here is output:
```
2017-11-13 18:38:55: netdata INFO : PLUGINSD: '/usr/libexec/netdata/plugins.d/node.d.plugin' running on pid 27900
2017-11-13 18:38:55: node.d.plugin: ERROR: Cannot load module's stiebeleltron config from /etc/netdata/node.d/stiebeleltron.conf exception: Error: ENOENT: no such file or directory, open '/etc/netdata/node.d/stiebeleltron.conf', using internal defaults.
2017-11-13 18:38:56: node.d.plugin: ERROR: Cannot load module's sma_webbox config from /etc/netdata/node.d/sma_webbox.conf exception: Error: ENOENT: no such file or directory, open '/etc/netdata/node.d/sma_webbox.conf', using internal defaults.
2017-11-13 18:38:56: node.d.plugin: ERROR: Cannot load module's named config from /etc/netdata/node.d/named.conf exception: Error: ENOENT: no such file or directory, open '/etc/netdata/node.d/named.conf', using internal defaults.
2017-11-13 18:38:56: node.d.plugin: ERROR: Cannot load module's fronius config from /etc/netdata/node.d/fronius.conf exception: Error: ENOENT: no such file or directory, open '/etc/netdata/node.d/fronius.conf', using internal defaults.

```

**Comment by ktsaou** [comment]
I think the fix I submitted above, did not fix the problem:

```
2017-11-13 18:33:38: node.d.plugin: DEBUG: named: configured, reporting 1 eligible services
```

Instead of disabling `named`, it enabled it.
looking into it...

**Comment by talkyy** [comment]
But the fix generates charts in dashboard. Do you have any snmp device to test? If you need more informations, let ask me.

**Comment by ktsaou** [comment]
yeah... the problem is not snmp. It the `named` plugin. you have something listening on port 8888, which is used by the named plugin to autodetect an isc-bind named server. The response it gets back, confuses it and it crashes. So, I am just attempting to gracefully handle the error.

SNMP works, but because `named` does not handle the issue properly, snmp is not proceeding.

I think the new PR will fix the issue.

**Comment by ktsaou** [comment]
ok, I merged it.
Could you please update and retry?

**Comment by talkyy** [comment]
Oh, I think I know the cause of this problem. I installed Chronograf and it works on port 8888. I think this makes conflicts. I have just stopped chronograf service and update netdata, here is result:
![image](https://user-images.githubusercontent.com/26269505/32739326-149a4c4e-c8d2-11e7-9e0c-0cf8f12708fb.png)
I will add more port and if I have problems, I hope you don't mind when I ask for your help.
Thank you so much.
